src/uu/install/src/install.rs

```diff
@@ -466,6 +489,12 @@ fn directory(paths: &[String], b: &Behavior) -> UResult<()> {
                 uucore::error::set_exit_code(1);
                 continue;
             }
+
+            if chown_optional_user_group(path, b).is_err() {
+                // Error messages are printed by the chown_optional_user_group function!
```

Comment:
This is not entirely true. The `InstallError::MetadataFailed` is not printed. I'd make `chown_optional_user_group` not print any errors and then use the [`show_if_err!`](https://docs.rs/uucore/latest/uucore/macro.show_if_err.html) macro here:
```rust
show_if_err!(chown_optional_user_group(path, b));
```
Alternatively, if you want to be a bit more explicit:
```rust
if Err(e) = chown_optional_user_group(path, b) {
    show!(e);
}
```
(these are equivalent, because that's what `show_if_err` expands to)

`show!` will print the message and set the exit code.

Reply:
Followed your advice and added an `InstallError::ChownFailed` with the error message. Please take a look.